alicloud/resource_alicloud_ram_account_password_policy.go

```diff
@@ -0,0 +1,222 @@
+var (
+	default_minimum_password_length      = 12
+	default_require_lowercase_characters = true
+	default_require_uppercase_characters = true
+	default_require_numbers              = true
+	default_require_symbols              = true
+	default_hard_expiry                  = false
+	default_max_password_age             = 0 // means disable
+	default_password_reuse_prevention    = 0 // means disable
+	default_max_login_attempts           = 5
+)
+
+func resourceAlicloudRamAccountPasswordPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudRamAccountPasswordPolicyUpdate,
+		Read:   resourceAlicloudRamAccountPasswordPolicyRead,
+		Update: resourceAlicloudRamAccountPasswordPolicyUpdate,
+		Delete: resourceAlicloudRamAccountPasswordPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"minimum_password_length": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      default_minimum_password_length,
+				ValidateFunc: intBetween(8, 32),
+			},
+			"require_lowercase_characters": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  default_require_lowercase_characters,
+			},
+			"require_uppercase_characters": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  default_require_uppercase_characters,
+			},
+			"require_numbers": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  default_require_numbers,
+			},
+			"require_symbols": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  default_require_symbols,
+			},
+			"hard_expiry": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  default_hard_expiry,
+			},
+			"max_password_age": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      default_max_password_age,
+				ValidateFunc: intBetween(0, 1095),
+			},
+			"password_reuse_prevention": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      default_password_reuse_prevention,
+				ValidateFunc: intBetween(0, 24),
+			},
+			"max_login_attempts": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      default_max_login_attempts,
+				ValidateFunc: intBetween(0, 32),
+			},
+		},
+	}
+}
+
+func resourceAlicloudRamAccountPasswordPolicyUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	request := ram.CreateSetPasswordPolicyRequest()
+	if v, ok := d.GetOk("minimum_password_length"); ok {
+		request.MinimumPasswordLength = requests.NewInteger(v.(int))
+	}
+	// due to lack of GetOkExists() in old Terraform included in vendor dir
+	// GetOk() doesn't distinguish beetwen false/0 and empty, so in else clause set 0 or false
+	if v, ok := d.GetOk("require_lowercase_characters"); ok {
+		request.RequireLowercaseCharacters = requests.NewBoolean(v.(bool))
+	} else {
```

Comment:
What do you think if we don't have `else` in all update? Maybe it's not necessary.

Reply:
Thanks for the advice, everything is always set, so I can change from `GetOk()` to `Get()` and streamline logic